nlmod/gwf/output.py

```diff
@@ -414,29 +411,19 @@ def get_head_at_point(
         Drop layers that are NaN at all timesteps. The default is True.
     rotated : bool, optional
         If the model grid has a rotation, and rotated is False, x and y are in model
-        coordinates. Otherwise x and y are in real world coordinates. The defaults is
-        False.
+        coordinates. Otherwise x and y are in real world coordinates. The default is
+        True.
 
     Returns
     -------
     head_point : xarray.DataArray
         A DataArray with dimensions (time, layer).
     """
-    if rotated and "angrot" in ds.attrs and ds.attrs["angrot"] != 0.0:
-        # calculate model coordinates from the specified real-world coordinates
-        x, y = get_affine_world_to_mod(ds) * (x, y)
-
     if "icell2d" in head.dims:
-        if gi is None:
-            if ds is None:
-                raise (
-                    ValueError(
-                        "Please supply either gi (GridIntersect) or ds for a vertex grid"
-                    )
-                )
-            gi = flopy.utils.GridIntersect(modelgrid_from_ds(ds), method="vertex")
-        icelld2 = gi.intersect(Point(x, y))["cellids"][0]
-        head_point = head[:, :, icelld2]
+        if ds is None:
+            raise (ValueError("Please supply ds for a vertex grid"))
+        icell2d = get_icell2d_from_xy(x, y, ds=ds, gi=gi, rotated=rotated)
+        head_point = head.sel(icell2d=icell2d)
     else:
         head_point = head.interp(x=x, y=y, method="nearest")
```

Comment:
This isn't guaranteerd to give you the head in the correct cell...

Reply:
You mean the last line, that uses interp? You are right...

Reply:
Yea, not something you touched in this PR, but I noticed :)